### DISCUSSION TITLE
Trying to understand the behavior of attitude rate mode

### DISCUSSION THREAD

**Discussion by gy2256** [bitcraze/discussions#1205] [Q&A]
Hi, I am trying to use attitude rate controller by changing the following line from `ANGLE` to `RATE`. 

https://github.com/bitcraze/crazyflie-firmware/blob/f61da11d54b6d54c7fa746688e8e9be4edd73a29/src/modules/src/crtp_commander_rpyt.c#L70-L72

After digging into the source code a bit, I notice the following
https://github.com/bitcraze/crazyflie-firmware/blob/f61da11d54b6d54c7fa746688e8e9be4edd73a29/src/modules/src/crtp_commander_rpyt.c#L186-L189. 

Any idea why `setpoint->attitude.roll = 0` is in the attitude rate mode? I thought this mode does not care about the absolute angle. The drone should behave more like acro mode in FPV drone flying.

**Comment by knmcguire** [comment]
Hi! I've converted your question to a discussion, but I do see that the code previews don't work...

But this is a good question! It doesn't seem logical. Have you tried removing it an see what happend?

**Reply by gy2256** [reply]
Yes, I conducted a quick test yesterday. After flashing the firmware with `RATE` enabled, the drone did not behave as expected. Then I removed those lines but the result remained unchanged. In both scenarios, the propellers are spinning at a low speed, and it does not have enough lift to take off. I tried changing the values of roll_rate, pitch_rate, and yaw_rate, but it didn't make any difference. I did test out the  `ANGLE` mode and it worked perfectly, so it's not a hardware issue. Currently, I'm investigating the potential issues.

**Reply by knmcguire** [reply]
Hmmm that's odd, but on the other hand the rate attitude controller is not something we test our regularly... I'll make an issue/ticket for this but I'm afraid that it might be something we won't put a very high priority on solving soon. But do keep up us up to date with anything else you are able to find.

**Comment by gy2256** [comment]
I have finished the debugging process. The slow-spinning propeller issue was not caused by the firmware; it's due to an implementation error we had when we were using a custom ROS message. I can confirm that the rate controller indeed works after removing the `setpoint->attitude.roll = 0`. It's unclear to me why that line is under the rate controller mode 🧐

**Reply by knmcguire** [reply]
thanks for testing this out! and yes this remains a mystery... we can probably just leave it there as it does not do any harm :)
